src/NServiceBus.Core/Pipeline/BehaviorContext.cs

```diff
@@ -8,16 +9,21 @@
     /// </summary>
     public abstract class BehaviorContext : ContextBag
     {
+        //TODO DanielTim: Does it make to force us to provide a RootContext here (to ensure Settings are set)? RootContext currently is private.
         /// <summary>
         /// Create an instance of <see cref="BehaviorContext"/>.
         /// </summary>
         /// <param name="parentContext">The parent context.</param>
-        protected BehaviorContext(ContextBag parentContext) : base(parentContext)
+        protected BehaviorContext(BehaviorContext parentContext) : base(parentContext)
         {
-          
         }
 
         /// <summary>
+        /// The readonly settings.
+        /// </summary>
+        public ReadOnlySettings Settings => Get<ReadOnlySettings>();
```

Comment:
Fair enough. That would basically explode that PR. Fact is for BusOperations we need ReadOnlySettings right now.

Are you saying we should bring that "Locator" concept in with this PR and remove the aggregation of readonly settings?